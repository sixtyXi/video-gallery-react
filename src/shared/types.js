// @flow
import type { RecordFactory, RecordOf } from 'immutable';
import { Record } from 'immutable';

type MovieProps = {
  id: string | number,
  poster_path: string,
  title: string,
  genres: Array<string>,
  release_date: string,
  vote_average: number,
  tagline: string,
  runtime: number,
  overview: string
};

const defaultValuesMovie: MovieProps = {
  id: '',
  poster_path: '',
  title: '',
  genres: [],
  release_date: '',
  vote_average: 0,
  tagline: '',
  runtime: 0,
  overview: ''
};

export type Movie = RecordOf<MovieProps>;

export const makeMovie: RecordFactory<Movie> = Record(defaultValuesMovie);

export type Filter = {
  value: string,
  title: string
};

export type MovieFilters = {
  searchBy: string,
  sortBy: string,
  search: string
};

export type Location = {
  search: string,
  pathname: string
};
