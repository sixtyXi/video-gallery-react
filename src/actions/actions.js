import { moviesAPI } from '../services/moviesAPI';

export const SET_MOVIE_LIST = 'SET_MOVIE_LIST';
export const SET_SEARCH_BY = 'SET_SEARCH_BY';
export const SET_SORT_BY = 'SET_SORT_BY';
export const SET_SEARCH = 'SET_SEARCH';
export const SET_MOVIE = 'SET_MOVIE';
export const SET_MOVIES_BY_GENRE = 'SET_MOVIES_BY_GENRE';

export const setMovieList = movies => {
  return {
    type: SET_MOVIE_LIST,
    payload: {
      movies
    }
  };
};

export const setMovie = movie => {
  return {
    type: SET_MOVIE,
    payload: {
      movie
    }
  };
};

export const setMoviesByGenre = movies => {
  return {
    type: SET_MOVIES_BY_GENRE,
    payload: {
      movies
    }
  };
};

export const fetchMovieList = () => (dispatch, getState) => {
  const { searchBy, sortBy, search } = getState().movieList;

  return moviesAPI
    .getMovies({ searchBy, sortBy, search })
    .then(movies => dispatch(setMovieList(movies)));
};

export const fetchMovieListByGenre = genres => dispatch => {
  return moviesAPI.getMoviesByGenre(genres).then(movies => {
    dispatch(setMoviesByGenre(movies));
  });
};

export const fetchMovie = id => dispatch => {
  return moviesAPI.getMovie(id).then(movie => dispatch(setMovie(movie)));
};

export const setSearchBy = filterValue => {
  return {
    type: SET_SEARCH_BY,
    payload: {
      filterValue
    }
  };
};

export const setSortBy = filterValue => {
  return {
    type: SET_SORT_BY,
    payload: {
      filterValue
    }
  };
};

export const setSearch = searchValue => {
  return {
    type: SET_SEARCH,
    payload: {
      searchValue
    }
  };
};
