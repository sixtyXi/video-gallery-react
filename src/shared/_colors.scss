$accent: #fd4d61;
$secondary: #414141;
