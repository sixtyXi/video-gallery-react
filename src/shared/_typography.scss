$text-light: white;
$size-m: 14px;
$size-s: 12px;
$size-xs: 10px;
$size-l: 16px;
