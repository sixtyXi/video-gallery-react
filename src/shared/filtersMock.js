export const SORT_FILTERS = [
  {
    value: 'release_date',
    title: 'release date'
  },
  {
    value: 'vote_average',
    title: 'rating'
  }
];

export const SEARCH_FILTERS = [
  {
    value: 'title',
    title: 'title'
  },
  {
    value: 'genres',
    title: 'genre'
  }
];

export const DEFAULT_FILTERS = {
  searchBy: SEARCH_FILTERS[0].value,
  sortBy: SORT_FILTERS[0].value,
  search: ''
};
