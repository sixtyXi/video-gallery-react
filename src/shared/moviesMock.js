export const MOVIES = [
  {
    id: 680,
    title: 'Pulp Fiction',
    tagline: "Just because you are a character doesn't mean you have character.",
    vote_average: 8.3,
    vote_count: 10910,
    release_date: '1994-09-10',
    poster_path: 'https://image.tmdb.org/t/p/w500/dM2w364MScsjFf8pfMbaWUcWrR.jpg',
    overview:
      "A burger-loving hit man, his philosophical partner, a drug-addled gangster's moll and a washed-up boxer converge in this sprawling, comedic crime caper. Their adventures unfurl in three stories that ingeniously trip back and forth in time.",
    budget: 8000000,
    revenue: 213928762,
    genres: ['Thriller', 'Crime'],
    runtime: 154
  },
  {
    id: 2,
    title: 'Kill Bill: Vol. 1',
    tagline: 'Oscar-winning Movies',
    vote_average: 4.5,
    release_date: '2003',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/71XwrZLWOBL._SY450_.jpg',
    overview: `From groundbreaking writer and director Quentin Tarantino KILL BILL VOLUME 1 stars Uma Thurman, Lucy Liu, and Vivica A. Fox in an epic vendetta! Four years after taking a bullet in the head at her own wedding, The Bride emerges from a coma and decides it's time for payback! Having been gunned down by her former boss and his deadly squad of international assassins, it's a kill-or- be- killed fight.`,
    runtime: 150,
    genres: ['action', 'adventure']
  },
  {
    id: 3,
    title: 'Four rooms',
    tagline: 'Oscar-winning Movies',
    vote_average: 3,
    release_date: '1995',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/51qcn-rYWqL._SY450_.jpg',
    overview: `Don't miss the fun in this hilariously sexy comedy that has Antonio Banderas (Once Upon a Time in Mexico), Madonna (Swept Away), and a sizzling all-star cast checking in for laughs! It's Ted the Bellhop's (Tim Roth, Pulp Fiction) first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments.`,
    runtime: 97,
    genres: ['comedies']
  },
  {
    id: 4,
    title: 'Kill Bill: Vol. 1',
    tagline: 'Oscar-winning Movies',
    vote_average: 4.5,
    release_date: '2003',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/71XwrZLWOBL._SY450_.jpg',
    overview: `From groundbreaking writer and director Quentin Tarantino KILL BILL VOLUME 1 stars Uma Thurman, Lucy Liu, and Vivica A. Fox in an epic vendetta! Four years after taking a bullet in the head at her own wedding, The Bride emerges from a coma and decides it's time for payback! Having been gunned down by her former boss and his deadly squad of international assassins, it's a kill-or- be- killed fight.`,
    runtime: 150,
    genres: ['action', 'adventure']
  },
  {
    id: 5,
    title: 'Four rooms',
    tagline: 'Oscar-winning Movies',
    vote_average: 3,
    release_date: '1995',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/51qcn-rYWqL._SY450_.jpg',
    overview: `Don't miss the fun in this hilariously sexy comedy that has Antonio Banderas (Once Upon a Time in Mexico), Madonna (Swept Away), and a sizzling all-star cast checking in for laughs! It's Ted the Bellhop's (Tim Roth, Pulp Fiction) first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments.`,
    runtime: 97,
    genres: ['comedies']
  },
  {
    id: 6,
    title: 'Kill Bill: Vol. 1',
    tagline: 'Oscar-winning Movies',
    vote_average: 4.5,
    release_date: '2003',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/71XwrZLWOBL._SY450_.jpg',
    overview: `From groundbreaking writer and director Quentin Tarantino KILL BILL VOLUME 1 stars Uma Thurman, Lucy Liu, and Vivica A. Fox in an epic vendetta! Four years after taking a bullet in the head at her own wedding, The Bride emerges from a coma and decides it's time for payback! Having been gunned down by her former boss and his deadly squad of international assassins, it's a kill-or- be- killed fight.`,
    runtime: 150,
    genres: ['action', 'adventure']
  },
  {
    id: 7,
    title: 'Four rooms',
    tagline: 'Oscar-winning Movies',
    vote_average: 3,
    release_date: '1995',
    poster_path: 'https://images-na.ssl-images-amazon.com/images/I/51qcn-rYWqL._SY450_.jpg',
    overview: `Don't miss the fun in this hilariously sexy comedy that has Antonio Banderas (Once Upon a Time in Mexico), Madonna (Swept Away), and a sizzling all-star cast checking in for laughs! It's Ted the Bellhop's (Tim Roth, Pulp Fiction) first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments.`,
    runtime: 97,
    genres: ['comedies']
  }
];
