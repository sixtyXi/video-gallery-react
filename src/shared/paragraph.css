.paragraph {
  height: 50px;
  display: inline-block;
  padding: 10px;
  line-height: 30px;
}
