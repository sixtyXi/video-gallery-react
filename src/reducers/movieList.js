// @flow

import { fromJS } from 'immutable';
import type { IndexedCollection, Map } from 'immutable';

import {
  SET_MOVIE_LIST,
  SET_SEARCH_BY,
  SET_SORT_BY,
  SET_SEARCH,
  SET_INITIAL_STATE
} from '../actions/actions';
import { DEFAULT_FILTERS } from '../shared/filtersMock';

import type { MovieFilters } from '../shared/types';

export type MovieListState = MovieFilters & {
  movies: IndexedCollection<Map<string, any>>
};

export const initialState: MovieListState = fromJS({
  movies: [],
  ...DEFAULT_FILTERS
});

export const movieList = (state: MovieListState = initialState, action: Function) => {
  switch (action.type) {
    case SET_MOVIE_LIST:
      return state.setIn(['movies'], fromJS(action.payload.movies));
    case SET_SEARCH_BY:
      return state.setIn(['searchBy'], action.payload.filterValue);
    case SET_SORT_BY:
      return state.setIn(['sortBy'], action.payload.filterValue);
    case SET_SEARCH:
      return state.setIn(['search'], action.payload.searchValue);
    case SET_INITIAL_STATE:
      return initialState;
    default:
      return state;
  }
};
