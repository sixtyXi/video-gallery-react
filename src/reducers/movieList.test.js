import { movieList as reducer, initialState } from './movieList';
import { SET_MOVIE_LIST, SET_SEARCH_BY, SET_SORT_BY, SET_SEARCH } from '../actions/actions';
import { MOVIES } from '../shared/moviesMock';
import { SEARCH_FILTERS, SORT_FILTERS } from '../shared/filtersMock';

describe('movieList reducer', () => {
  test('should return the initial state', () => {
    expect(reducer(initialState, {})).toMatchSnapshot();
  });

  test('should handle SET_MOVIE_LIST', () => {
    expect(
      reducer(initialState, {
        type: SET_MOVIE_LIST,
        payload: {
          movies: MOVIES
        }
      })
    ).toMatchSnapshot();
  });

  test('should handle SET_SEARCH_BY', () => {
    expect(
      reducer(initialState, {
        type: SET_SEARCH_BY,
        payload: {
          filterValue: SEARCH_FILTERS[1].value
        }
      })
    ).toMatchSnapshot();
  });

  test('should handle SET_SORT_BY', () => {
    expect(
      reducer(initialState, {
        type: SET_SORT_BY,
        payload: {
          filterValue: SORT_FILTERS[1].value
        }
      })
    ).toMatchSnapshot();
  });

  test('should handle SET_SEARCH', () => {
    const searchValue = 'cats';

    expect(
      reducer(initialState, {
        type: SET_SEARCH,
        payload: {
          searchValue
        }
      })
    ).toMatchSnapshot();
  });
});
