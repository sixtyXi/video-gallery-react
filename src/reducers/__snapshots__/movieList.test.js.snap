// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movieList reducer should handle SET_MOVIE_LIST 1`] = `
Immutable.Map {
  "movies": Immutable.List [
    Immutable.Map {
      "budget": 8000000,
      "release_date": "1994-09-10",
      "vote_average": 8.3,
      "tagline": "Just because you are a character doesn't mean you have character.",
      "genres": Immutable.List [
        "Thriller",
        "Crime",
      ],
      "poster_path": "https://image.tmdb.org/t/p/w500/dM2w364MScsjFf8pfMbaWUcWrR.jpg",
      "title": "Pulp Fiction",
      "runtime": 154,
      "overview": "A burger-loving hit man, his philosophical partner, a drug-addled gangster's moll and a washed-up boxer converge in this sprawling, comedic crime caper. Their adventures unfurl in three stories that ingeniously trip back and forth in time.",
      "vote_count": 10910,
      "id": 680,
      "revenue": 213928762,
    },
    Immutable.Map {
      "release_date": "2003",
      "vote_average": 4.5,
      "tagline": "Oscar-winning Movies",
      "genres": Immutable.List [
        "action",
        "adventure",
      ],
      "poster_path": "https://images-na.ssl-images-amazon.com/images/I/71XwrZLWOBL._SY450_.jpg",
      "title": "Kill Bill: Vol. 1",
      "runtime": 150,
      "overview": "From groundbreaking writer and director Quentin Tarantino KILL BILL VOLUME 1 stars Uma Thurman, Lucy Liu, and Vivica A. Fox in an epic vendetta! Four years after taking a bullet in the head at her own wedding, The Bride emerges from a coma and decides it's time for payback! Having been gunned down by her former boss and his deadly squad of international assassins, it's a kill-or- be- killed fight.",
      "id": 2,
    },
    Immutable.Map {
      "release_date": "1995",
      "vote_average": 3,
      "tagline": "Oscar-winning Movies",
      "genres": Immutable.List [
        "comedies",
      ],
      "poster_path": "https://images-na.ssl-images-amazon.com/images/I/51qcn-rYWqL._SY450_.jpg",
      "title": "Four rooms",
      "runtime": 97,
      "overview": "Don't miss the fun in this hilariously sexy comedy that has Antonio Banderas (Once Upon a Time in Mexico), Madonna (Swept Away), and a sizzling all-star cast checking in for laughs! It's Ted the Bellhop's (Tim Roth, Pulp Fiction) first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments.",
      "id": 3,
    },
    Immutable.Map {
      "release_date": "2003",
      "vote_average": 4.5,
      "tagline": "Oscar-winning Movies",
      "genres": Immutable.List [
        "action",
        "adventure",
      ],
      "poster_path": "https://images-na.ssl-images-amazon.com/images/I/71XwrZLWOBL._SY450_.jpg",
      "title": "Kill Bill: Vol. 1",
      "runtime": 150,
      "overview": "From groundbreaking writer and director Quentin Tarantino KILL BILL VOLUME 1 stars Uma Thurman, Lucy Liu, and Vivica A. Fox in an epic vendetta! Four years after taking a bullet in the head at her own wedding, The Bride emerges from a coma and decides it's time for payback! Having been gunned down by her former boss and his deadly squad of international assassins, it's a kill-or- be- killed fight.",
      "id": 4,
    },
    Immutable.Map {
      "release_date": "1995",
      "vote_average": 3,
      "tagline": "Oscar-winning Movies",
      "genres": Immutable.List [
        "comedies",
      ],
      "poster_path": "https://images-na.ssl-images-amazon.com/images/I/51qcn-rYWqL._SY450_.jpg",
      "title": "Four rooms",
      "runtime": 97,
      "overview": "Don't miss the fun in this hilariously sexy comedy that has Antonio Banderas (Once Upon a Time in Mexico), Madonna (Swept Away), and a sizzling all-star cast checking in for laughs! It's Ted the Bellhop's (Tim Roth, Pulp Fiction) first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments.",
      "id": 5,
    },
    Immutable.Map {
      "release_date": "2003",
      "vote_average": 4.5,
      "tagline": "Oscar-winning Movies",
      "genres": Immutable.List [
        "action",
        "adventure",
      ],
      "poster_path": "https://images-na.ssl-images-amazon.com/images/I/71XwrZLWOBL._SY450_.jpg",
      "title": "Kill Bill: Vol. 1",
      "runtime": 150,
      "overview": "From groundbreaking writer and director Quentin Tarantino KILL BILL VOLUME 1 stars Uma Thurman, Lucy Liu, and Vivica A. Fox in an epic vendetta! Four years after taking a bullet in the head at her own wedding, The Bride emerges from a coma and decides it's time for payback! Having been gunned down by her former boss and his deadly squad of international assassins, it's a kill-or- be- killed fight.",
      "id": 6,
    },
    Immutable.Map {
      "release_date": "1995",
      "vote_average": 3,
      "tagline": "Oscar-winning Movies",
      "genres": Immutable.List [
        "comedies",
      ],
      "poster_path": "https://images-na.ssl-images-amazon.com/images/I/51qcn-rYWqL._SY450_.jpg",
      "title": "Four rooms",
      "runtime": 97,
      "overview": "Don't miss the fun in this hilariously sexy comedy that has Antonio Banderas (Once Upon a Time in Mexico), Madonna (Swept Away), and a sizzling all-star cast checking in for laughs! It's Ted the Bellhop's (Tim Roth, Pulp Fiction) first night on the job...and the hotel's very unusual guests are about to place him in some outrageous predicaments.",
      "id": 7,
    },
  ],
  "searchBy": "title",
  "sortBy": "release_date",
  "search": "",
}
`;

exports[`movieList reducer should handle SET_SEARCH 1`] = `
Immutable.Map {
  "movies": Immutable.List [],
  "searchBy": "title",
  "sortBy": "release_date",
  "search": "cats",
}
`;

exports[`movieList reducer should handle SET_SEARCH_BY 1`] = `
Immutable.Map {
  "movies": Immutable.List [],
  "searchBy": "genres",
  "sortBy": "release_date",
  "search": "",
}
`;

exports[`movieList reducer should handle SET_SORT_BY 1`] = `
Immutable.Map {
  "movies": Immutable.List [],
  "searchBy": "title",
  "sortBy": "vote_average",
  "search": "",
}
`;

exports[`movieList reducer should return the initial state 1`] = `
Immutable.Map {
  "movies": Immutable.List [],
  "searchBy": "title",
  "sortBy": "release_date",
  "search": "",
}
`;
